lib/vector-config/src/stdlib.rs

```diff
@@ -21,51 +22,53 @@ use crate::{
 };
 
 // Unit type.
-impl<'de> Configurable<'de> for () {
-    fn generate_schema(_: &mut SchemaGenerator, _: Metadata<'de, Self>) -> SchemaObject {
+impl Configurable for () {
+    fn generate_schema(_: &mut SchemaGenerator, _: Metadata<Self>) -> SchemaObject {
         panic!("unit fields are not supported and should never be used in `Configurable` types");
     }
 }
 
 // Null and boolean.
-impl<'de, T> Configurable<'de> for Option<T>
+impl<T> Configurable for Option<T>
 where
-    T: Configurable<'de>,
+    T: Configurable + Serialize,
 {
     fn is_optional() -> bool {
         true
     }
 
-    fn metadata() -> Metadata<'de, Self> {
+    fn metadata() -> Metadata<Self> {
         // We clone the default metadata of the wrapped type because otherwise this "level" of the schema would
         // effective sever the link between things like the description of `T` itself and what we show for a field of
         // type `Option<T>`.
         //
         // To wit, this allows callers to use `#[configurable(derived)]` on a field of `Option<T>` so long as `T` has a
         // description, and both the optional field and the schema for `T` will get the description... but the
         // description for the optional field can still be overridden independently, etc.
-        T::metadata().map_default_value(|default| Some(default))
+        T::metadata().convert()
     }
 
-    fn generate_schema(gen: &mut SchemaGenerator, overrides: Metadata<'de, Self>) -> SchemaObject {
-        let inner_metadata = overrides.clone().flatten_default();
+    fn generate_schema(gen: &mut SchemaGenerator, overrides: Metadata<Self>) -> SchemaObject {
+        let mut inner_metadata = T::metadata();
+        inner_metadata.set_transparent();
```

Comment:
The rules are slightly complex, but the process of generating the schema for a given type happens in two parts: generating the base level schema, and then applying the metadata which is where we set default values, validation, etc.

Similar to the comments above about not using the default value of `T` for `Option<T>`, we're handling the reverse case here. The previous code was taking the metadata for the `Option<T>` -- this will always be an actual field in a struct/enum -- which may have contained a default value, and then using that when generating the schema for `T`.

If someone has a `Option<T>` field with a default value, it doesn't make sense to generate the schema for `T` while saying the default value for it is whatever was on that optional field... because there might be 1 or 100 other fields that are also `Option<T>`, with different defaults, and now our schema is implying that `T` by itself always has a default of whatever the first `Option<T>` field we generated a schema for had for it's default value...

I feel like if I say `Option<T>` or "default value" one more time my brain is gonna explode, but I think you get the point here.